compute_tools/src/extension_server.rs

```diff
@@ -42,80 +51,62 @@ pub fn get_pg_version(pgbin: &str) -> String {
+    remote_from_path: RemotePath,
+    sub_directory: Option<&str>,
     download_location: &Path,
 ) -> anyhow::Result<()> {
-    // downloads file at remote_from_path to download_location/[file_name]
-
-    // we cannot use remote_from_path.object_name() here
-    // because extension files can be in subdirectories of the extension store.
-    //
-    // To handle this, we use remote_from_prefix to strip the prefix from the path
-    // this gives us the relative path of the file in the extension store,
-    // and we use this relative path to construct the local path.
-    //
-    let local_path = match remote_from_prefix {
-        Some(prefix) => {
-            let p = remote_from_path
-                .get_path()
-                .strip_prefix(prefix)
-                .expect("bad prefix");
-
-            download_location.join(p)
-        }
+    // downloads file at remote_from_path to
+    // `download_location/[optional: subdirectory]/[remote_storage.object_name()]`
+    // Note: the subdirectory commmand is needed when there is an extension that
+    // depends on files in a subdirectory.
+    // For example, v14/share/extension/some_ext.control
+    // might depend on v14/share/extension/some_ext/some_ext--1.1.0.sql
+    // and v14/share/extension/some_ext/xxx.csv
+    // Note: it is the caller's responsibility to create the appropriate subdirectory
+
+    let local_path = match sub_directory {
+        Some(subdir) => download_location
+            .join(subdir)
+            .join(remote_from_path.object_name().expect("bad object")),
         None => download_location.join(remote_from_path.object_name().expect("bad object")),
     };
-
     if local_path.exists() {
         info!("File {:?} already exists. Skipping download", &local_path);
         return Ok(());
     }
-
     info!(
         "Downloading {:?} to location {:?}",
         &remote_from_path, &local_path
     );
-    let mut download = remote_storage.download(remote_from_path).await?;
+    let mut download = remote_storage.download(&remote_from_path).await?;
     let mut write_data_buffer = Vec::new();
     download
         .download_stream
         .read_to_end(&mut write_data_buffer)
         .await?;
-    if remote_from_prefix.is_some() {
-        if let Some(prefix) = local_path.parent() {
-            info!(
-                "Downloading file with prefix. Create directory {:?}",
-                prefix
-            );
-            // if directory already exists, this is a no-op
-            std::fs::create_dir_all(prefix)?;
-        }
-    }
-
     let mut output_file = BufWriter::new(File::create(local_path)?);
     output_file.write_all(&write_data_buffer)?;
+    info!("Download {:?} completed successfully", &remote_from_path);
     Ok(())
 }
 
 // download extension control files
 //
-// if custom_ext_prefixes is provided - search also in custom extension paths
+// if custom_ext_prefixes is provided - search also in custom extension paths for private extensions
```

Comment:
we agreed to use term `custom` instead of `private` everywhere, so this is unneeded change

Reply:
ok, I reverted it